ДЗ номер 2 готова.

ikr0ni opened the pull request:
Без параметров выдает usage

Commit:
ДЗ номер 2 готова.

Без параметров выдает usage

v-lozhkin commented on lesson2/main.go:
```diff
@@ -0,0 +1,78 @@
+
+// Считаем диаметр и длинну окружности по площади круга
+func calcCircleDL(square int) (float64, float64) {
+	length := math.Pow(float64(square), 2) / (4 * math.Pi)
```
Формулы не совсем корректные.

ikr0ni replied:
Исправлено в commit 142f56063391e0dd0f4b7bff0d259ae24eacb022

v-lozhkin commented on lesson2/main.go:
```diff
@@ -0,0 +1,78 @@
+)
+
+// Считаем площадь прямоугольника
+func calcSquare(top int, left int) int {
```
Лайк за использование функций

v-lozhkin commented on lesson2/main.go:
```diff
@@ -0,0 +1,78 @@
+	}
+	switch *oper {
+	case "calcSquare":
+		{
```
Лайк за свитч, но фигурные скобки внутри кейсов не нужны

ikr0ni replied:
Исправлено в commit 142f56063391e0dd0f4b7bff0d259ae24eacb022

v-lozhkin commented on lesson2/main.go:
```diff
@@ -0,0 +1,78 @@
+			} else {
+				fmt.Println("Бей, круши, ломай")
+				hundreds, dozens, units := divideByDigits(*number)
+				fmt.Println("В числе", *number, "сотен", hundreds, "десятков", dozens, "единиц", units)
```
Вывод считывается странно %)
`В числе 999 сотен 9 десятков 9 единиц 9`
Сначала упорно не мог понять, откуда тут 999 сотен

ikr0ni replied:
Исправлено в commit 142f56063391e0dd0f4b7bff0d259ae24eacb022

v-lozhkin commented on lesson2/main.go:
```diff
@@ -0,0 +1,78 @@
+func calcSquare(top int, left int) int {
+	square := top * left
+
+	return square
```
Можно возвращщать сразу умножение, не обязательно складывать результат в переменную:
```go
return top * left
```

ikr0ni replied:
Исправлено в commit 142f56063391e0dd0f4b7bff0d259ae24eacb022

v-lozhkin commented on lesson2/main.go:
```diff
@@ -0,0 +1,78 @@
+func divideByDigits(number int) (int, int, int) {
+	hundreds := number / 100
+	dozens := (number % 100) / 10
+	units := ((number % 100) % 10)
```
Лишние внешние скобки

ikr0ni replied:
Исправлено в commit 142f56063391e0dd0f4b7bff0d259ae24eacb022

Commit:
1. Строка вывода разделения доработана для более удобного восприятия.
2. Убраны лишние скобки в рассчете units
3. Функция calcSquare оптимизирована, обрано лишнее присвоение
4. Убраны лишние фигурные скобки в свитче
5. Формулы Диаметра и длинны окружности исправлены

v-lozhkin reviewed (approved):
Все супер
